ftw/zipextract/browser/zipextract_view.py

```diff
@@ -0,0 +1,71 @@
+from Products.Five.browser import BrowserView
+from ftw.zipextract.zipextracter import ZipExtracter
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from plone.i18n.normalizer import idnormalizer
+import os
+from z3c.form import form
+
+class ZipExtractView(form.Form):
+
+    template = ViewPageTemplateFile('templates/zipextract.pt')
+
+    def __call__(self):
+        self.zipextracter = ZipExtracter(self.context)
+        self.build_tree_html()
+        if self.request.get('form.submitted'):
+            self.unzip(self.request)
+            return self.redirect_to_workspace()
+        return self.template()
+
+    @staticmethod
+    def _file_size_repr(file_node):
+        size = file_node.info.file_size
+        if size > 1024:
+            return str(size/1024)+"KB"
+        else:
+            return str(size)+"B"
+
+    def _recurse_tree_html(self, folder_node):
+        for file_node in folder_node.file_dict.values():
+            label = file_node.name + ": " + self._file_size_repr(file_node)
+            self.tree_html += '<li id="{id}" class="file"><input type="checkbox" value="{path}"\
+                id="{id}_checkbox" name="files:list" /><label for="{id}_checkbox">{name}</label></li>\n'\
```

Comment:
I'd try to move rendering to a template. Doing it like this is prone to prone for XSS-bugs.

Reply:
done